database/postgres.go

```diff
@@ -135,11 +136,25 @@ func PostgresUniqueViolation(err error) bool {
 	if err == nil {
 		return false
 	}
+	match := strings.Contains(err.Error(), postgresErrUniqueViolation)
+
 	var pgError *pgconn.PgError
 	if errors.As(err, &pgError) {
-		if pgError.Code == postgresErrUniqueViolation {
-			return true
-		}
+		return match || pgError.Code == postgresErrUniqueViolation
```

Comment:
I think you can remove the `match ||` here -- cause that'll be covered by the `return match` below?

Reply:
No, if you remove `match ||` then `pgError.Code == postgresErrUniqueViolation` would return if it evaluates to false.

That return could be too eager if `match` evaluates to true.

Reply:
mm i see, was mainly wondering if we could remove the `match` out of the `errors.As(err, &pgError` block -- think you could simplify this to:

```go
if errors.As(err, &pgError) && 
   pgError.Code == postgresErrUniqueViolation {
   return true
}

return strings.Contains(err.Error(), postgresErrUniqueViolation)
```

Reply:
updated and added more tests [564f288](https://github.com/moov-io/base/pull/448/commits/564f2880d8f4d4d5ad76530f1d68ccf63024380b)